arrayfire/array.py

```diff
@@ -463,13 +461,8 @@ def __init__(self, src=None, dims=None, dtype=None, is_device=False, offset=None
         buf = None
         buf_len = 0
 
-        if dtype is not None:
-            if isinstance(dtype, str):
-                type_char = dtype
-            else:
-                type_char = to_typecode[dtype.value]
-        else:
```

Comment:
This branch won't happen, error on line 499 follows

Reply:
fixed